src/Startup/Program.cs

```diff
@@ -94,35 +95,39 @@ public Program(string[] args)
         /// The main method.
         /// </summary>
         /// <param name="args">The command line args.</param>
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
             var logRepository = LogManager.GetRepository(Assembly.GetEntryAssembly());
             XmlConfigurator.ConfigureAndWatch(logRepository, new FileInfo(Log4NetConfigFilePath));
 
-            using (new Program(args))
+            using var program = new Program(args);
+            using var exitCts = new CancellationTokenSource();
+            var exitToken = exitCts.Token;
+            void OnCancelKeyPress(object sender, ConsoleCancelEventArgs e)
             {
-                var exit = false;
+                exitCts.Cancel();
+                Console.CancelKeyPress -= OnCancelKeyPress;
```

Comment:
is that neccessary?

Reply:
It’s just to be sure that CancellationTokenSource.Cancel() isn’t called after it has been disposed. When the application wasn’t exited, I noticed that pressing Ctrl+C another time caused an exception and crash.

Reply:
oh, i just saw this crash. should keep this then. thnx

Reply:
this exception?
```
Unhandled exception. System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at MUnique.OpenMU.Startup.TextReaderExtensions.ReadLineAsync(TextReader reader, CancellationToken cancellationToken) in /home/ja0n/Projects/OpenMU/src/Startup/TextReaderExtensions.cs:line 26
   at MUnique.OpenMU.Startup.Program.Main(String[] args) in /home/ja0n/Projects/OpenMU/src/Startup/Program.cs:line 137
   at MUnique.OpenMU.Startup.Program.<Main>(String[] args)
```

Reply:
this exception?
```
Unhandled exception. System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at MUnique.OpenMU.Startup.TextReaderExtensions.ReadLineAsync(TextReader reader, CancellationToken cancellationToken)
```